packages/edit-site/src/utils/use-activate-theme.js

```diff
@@ -30,7 +33,9 @@ export function useActivateTheme() {
 				currentlyPreviewingTheme() +
 				'&_wpnonce=' +
 				window.WP_BLOCK_THEME_ACTIVATE_NONCE;
+			startResolution( 'activateTheme' );
 			await window.fetch( activationURL );
+			finishResolution( 'activateTheme' );
 			const { wp_theme_preview: themePreview, ...params } =
 				location.params;
 			history.replace( params );
```

Comment:
Outside the scope of this PR, but it's worth noting that `history.replace` (which removes `?wp_theme_preview=`) is executed regardless of the HTTP status returned by `window.fetch`, including a 500 error 😇  CC: @scruffian